File: Medium/0011_container-with-most-water/Solution.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        left = 0
        right = len(height) - 1
        max_val = 0

        while left < right:
            area = min(height[left], height[right]) * (right - left)

            if area > max_val:
                max_val = area

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_val

File: Medium/0011_container-with-most-water/Solution.cpp
#include <vector>
#include <algorithm>

class Solution {
public:
    int maxArea(std::vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int max_val = 0;

        while (left < right) {
            int area = std::min(height[left], height[right]) * (right - left);

            if (area > max_val) {
                max_val = area;
            }

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return max_val;
    }
};